### PR TITLE
Remove `frontsBannerAds` switch

### PR DESCRIPTION
**Pull request by domlander**
## What does this change?

- Removes the `frontsBannerAds` switch and redundant logic paths

## Why?

- We no longer need this switch: there is no intention to revert back to displaying MPU ads on fronts.

## Notes

We would like to remove the following components, but they may still be in use. For now, remove their ability to serve MPU ads:
- [DynamicSlowMPU](https://github.com/guardian/dotcom-rendering/blob/main/dotcom-rendering/src/components/DynamicSlowMPU.tsx)
- [FixedMediumSlowXIIMPU](https://github.com/guardian/dotcom-rendering/blob/main/dotcom-rendering/src/components/FixedMediumSlowXIIMPU.tsx)
- [FixedSmallSlowVMPU](https://github.com/guardian/dotcom-rendering/blob/main/dotcom-rendering/src/components/FixedSmallSlowVMPU.tsx)

## Follow up

Remove the [feature switch definition](https://github.com/guardian/frontend/blob/main/common/app/conf/switches/CommercialSwitches.scala#L162).

### CODE REVIEW

File: dotcom-rendering/src/lib/getTagFrontsAdPositions.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+			.map((collection) =>
+				collections.findIndex(
+					({ day, month, year }) =>
+						day === collection.day &&
+						month === collection.month &&
+						year === collection.year,
+				),
+			)
+			// Should insert no more than 10 ads
+			.slice(0, 10)
```
Comment (mxdvl) [inline dotcom-rendering/src/lib/getTagFrontsAdPositions.ts:45]: Maybe extract as `MAX_MOBILE_ADS`?

File: dotcom-rendering/src/lib/getTagFrontsAdPositions.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+					!hasAdjacentCommercialContainer(index, merchHighPosition),
+			)
+			.filter(isEvenIndex)
+			.map((collection) =>
+				collections.findIndex(
+					({ day, month, year }) =>
+						day === collection.day &&
+						month === collection.month &&
+						year === collection.year,
+				),
```
Comment (mxdvl) [inline dotcom-rendering/src/lib/getTagFrontsAdPositions.ts:42]: This [can technically return `-1`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/findIndex#return_value). Do you want to filter for it?

File: dotcom-rendering/src/lib/getFrontsAdPositions.ts [hunk 3]
```diff
@@ -25,7 +88,7 @@ const getCollectionHeight = (
 		DCRCollectionType,
 		'collectionType' | 'containerPalette' | 'grouped'
 	>,
-): number => {
+): 0.5 | 1 | 1.5 | 2 | 2.5 | 3 => {
```
Comment (mxdvl) [inline dotcom-rendering/src/lib/getFrontsAdPositions.ts:91]: <3

**Review by mxdvl — APPROVED** [review]
This seems generally fine, but would it be possible at all to split this into smaller PRs?

File: dotcom-rendering/src/components/FixedMediumSlowXIIMPU.tsx [hunk 4]
```diff
@@ -132,78 +120,10 @@ const Card50_Card50 = ({
...
-			<LI percentage="33.333%" padSides={true} showDivider={true}>
-				<Hide until="tablet">
-					<AdSlot position="inline" index={adIndex} />
-				</Hide>
-			</LI>
-		</UL>
-	);
-};
-
 export const FixedMediumSlowXIIMPU = ({
```
[the remark this answers is not in the document]
Comment (domlander) [reply dotcom-rendering/src/components/FixedMediumSlowXIIMPU.tsx:123]: I think there will need to be a follow up piece of work to tidy up these containers.

For example, the `FixedSmallSlowVMPU` container will only display IV cards when the MPU is removed. There exists a `FixedSmallSlowIV` container, which may now be the same thing. However, the `FixedMediumSlowXIIMPU` container doesn't seem to have a non-MPU version: there is no FixedMediumSlowXI container.

As you suggested in the chat, a joint WebX-Commercial follow-up effort to simplify these containers would be a good idea.

**Comment by domlander** [comment]
> This seems generally fine, butw ould it be possible at all to split this into smaller PRs?

Yeah I think this is a good idea. I'll add a separate PR for the refactor

**Pull request merged**